error: no field or member function named 'exec' in 'Build'

I have tested this on Linux and Windows and am seeing the same build error.

This is with https://github.com/fubark/zig-v8/commit/10654d4eed3e6fb4ebc5407577d779cc4235955a, which added support for ` 0.12.0-dev`, I'm a bit of a Zig newbie, so not sure how to fix this yet.
 @almmiko any idea?


**`zig version`** : `0.12.0-dev.1286+cc394431a` 


`zig build get-tools`
```
zig-v8\build.zig:639:23: error: no field or member function named 'exec' in 'Build'
            _ = self.b.exec(&.{ "git", "clone", "--depth=1", "--branch", v8_rev, "https://chromium.googlesource.com/v8/v8.git", "v8" });
                ~~~~~~^~~~~
C:\dev\zig-windows-x86_64-0.12.0-dev.1286+cc394431a\lib\std\Build.zig:1:1: note: struct declared here
const std = @import("std.zig");
^~~~~
referenced by:
    create: C:\dev\zig-v8\build.zig:568:27
    createGetV8: C:\dev\zig-v8\build.zig:355:36
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

> Hey @tritao zig no longer has the `exec` method. It was renamed to `run`. See [ziglang/zig@fd2239b](https://github.com/ziglang/zig/commit/fd2239bde9e2051a32fb25c50c732a40d4afccd0)

Thanks, yeah I forgot to update this. I found that commit earlier, though there is still another issue after this one is fixed:

```
/home/joao/dev/zig-linux-x86_64-0.12.0-dev.1270+6c9d34bce/lib/std/heap.zig:37:13: error: C allocator is only available when linking against libc
            @compileError("C allocator is only available when linking against libc");
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joao/dev/zig-linux-x86_64-0.12.0-dev.1270+6c9d34bce/lib/std/c/linux.zig:302:12: error: dependency on libc must be explicitly specified in the build command
pub extern "c" fn posix_memalign(memptr: *?*anyopaque, alignment: usize, size: usize) c_int;
           ^~~
/home/joao/dev/zig-linux-x86_64-0.12.0-dev.1270+6c9d34bce/lib/std/c/linux.zig:303:12: error: dependency on libc must be explicitly specified in the build command
pub extern "c" fn malloc_usable_size(?*const anyopaque) usize;
           ^~~
/home/joao/dev/zig-linux-x86_64-0.12.0-dev.1270+6c9d34bce/lib/std/c.zig:246:12: error: dependency on libc must be explicitly specified in the build command
pub extern "c" fn free(?*anyopaque) void;
```

Any idea? I can send a PR to fix this one.

--- a/build.zig
+++ b/build.zig
@@ -591,18 +591,23 @@ pub const GetV8SourceStep = struct {
 
         const stat = try statPathFromRoot(self.b, local_path);
         if (stat == .NotExist) {
-            _ = self.b.exec(&.{ "git", "clone", dep.repo_url, local_path });
+            _ = self.b.run(&.{ "git", "clone", dep.repo_url, local_path });
         }
-        _ = self.b.exec(&.{ "git", "-C", local_path, "checkout", dep.repo_rev });
+        _ = self.b.run(&.{ "git", "-C", local_path, "checkout", dep.repo_rev });
         if (stat == .NotExist) {
             // Apply patch for v8/build
             if (std.mem.eql(u8, key, "build")) {
-                _ = self.b.exec(&.{ "git", "apply", "--ignore-space-change", "--ignore-whitespace", "patches/v8_build.patch", "--directory=v8/build" });
+                _ = self.b.run(&.{ "git", "apply", "--ignore-space-change", "--ignore-whitespace", "patches/v8_build.patch", "--directory=v8/build" });
             }
         }
     }
 
     fn runHook(self: *Self, hooks: json.Value, name: []const u8) !void {
+        var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+        const alloc = gpa.allocator();
+
+        defer _ = gpa.deinit();
+
         for (hooks.array.items) |hook| {
             if (std.mem.eql(u8, name, hook.object.get("name").?.string)) {
                 const cmd = hook.object.get("action").?.array;
@@ -612,12 +617,11 @@ pub const GetV8SourceStep = struct {
                     try args.append(it.string);
                 }
                 const cwd = self.b.pathFromRoot("v8");
-                const alloc = std.heap.c_allocator;
 
                 const cmd_path = try std.fmt.allocPrint(alloc, "{s}/{s}", .{ cwd, args.items[1] });
                 defer alloc.free(cmd_path);
 
-                _ = self.b.exec(&.{ args.items[0], cmd_path });
+                _ = self.b.run(&.{ args.items[0], cmd_path });
                 break;
             }
         }
@@ -636,13 +640,13 @@ pub const GetV8SourceStep = struct {
         // Clone V8.
         const stat = try statPathFromRoot(self.b, "v8");
         if (stat == .NotExist) {
-            _ = self.b.exec(&.{ "git", "clone", "--depth=1", "--branch", v8_rev, "https://chromium.googlesource.com/v8/v8.git", "v8" });
+            _ = self.b.run(&.{ "git", "clone", "--depth=1", "--branch", v8_rev, "https://chromium.googlesource.com/v8/v8.git", "v8" });
             // Apply patch for v8 root.
-            _ = self.b.exec(&.{ "git", "apply", "--ignore-space-change", "--ignore-whitespace", "patches/v8.patch", "--directory=v8" });
+            _ = self.b.run(&.{ "git", "apply", "--ignore-space-change", "--ignore-whitespace", "patches/v8.patch", "--directory=v8" });
         }
 
         // Get DEPS in json.
-        const deps_json = self.b.exec(&.{ "python3", "parse_deps.py", "v8/DEPS" });
+        const deps_json = self.b.run(&.{ "python3", "parse_deps.py", "v8/DEPS" });
         defer self.b.allocator.free(deps_json);
 
         var tree = try json.parseFromSlice(json.Value, self.b.allocator, deps_json, .{});

